### ISSUE TITLE
no-implied-eval False positive

### ISSUE THREAD

**Issue by greyvugrin** [eslint/eslint#7821]
**Tell us about your environment**

* **ESLint Version:**
3.12.2
* **Node Version:**
6.9.1
* **npm Version:**
3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint  (7.1.1)

**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "rules": {
    "max-len": [1, 120, 2, {"ignoreComments": true}],
    "semi":["warn", "never"],
    "comma-dangle": ["error", {
        "functions": "never"
      }
    ],
    "no-underscore-dangle": [0],
    "react/prop-types": [0],
    "react/forbid-prop-types": [0],
    "react/jsx-filename-extension": [0],
    "import/no-named-as-default": [0],
    "import/no-extraneous-dependencies": ["error", {
        "devDependencies": ["**/*.spec.js"]
      }
    ],
    "jsx-a11y/no-static-element-interactions": [0],
    "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }]
  },
  "env": {
    "browser": true,
    "node": true,
    "mocha": true
  },
  "globals": {
    "__PROD__": false,
    "__DEV__": false
   ...
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function receiveProfileUpdate(user) {
  return (dispatch) => {
    dispatch(resetErrors())
    dispatch(setTimeoutableSuccessMessage('Successfully Updated Profile.'))
    dispatch(setCurrentUser(user))
    browserHistory.push(PROFILE_URL(user.username))
  }
}
```

**What did you expect to happen?**
No errors to be thrown

**What actually happened? Please include the actual, raw output from ESLint.**
`106:14  error  Implied eval. Consider passing a function instead of a string                                               no-implied-eval`

Caused by using a string as a parameter for `setTimeoutableSuccessMessage`

_Note:_

Updating code to : 
```js
function receiveProfileUpdate(user) {
  return (dispatch) => {
    dispatch(resetErrors())
    const message = 'Successfully Updated Profile.'
    dispatch(setTimeoutableSuccessMessage(message))
    dispatch(setCurrentUser(user))
    browserHistory.push(PROFILE_URL(user.username))
  }
}
```

fixed the issue, but it doesn't seem right to have to use that workaround. Now that I'm looking at it more the function name may be causing the rule to think it's `setTimeout`, and cause the bug.

**Comment by platinumazure** [comment]
I think you're right, we're checking for the function name starting with `setTimeout`. Definitely a bug. Thanks for reporting this!

**Comment by greyvugrin** [comment]
My pleasure!
